### ISSUE TITLE
Rebuilding indexes can fail with a NullReferenceException when trying to reindex forms data

### ISSUE THREAD

**Issue by Shazwazza** [umbraco/Umbraco.Forms.Issues#373]
Can occur during index rebuilding of the Forms index. The full error and stack trace is here. So some null checks need to be put somewhere in there

```
Failed to rebuild empty indexes.

Exception 
System.AggregateException: One or more errors occurred. ---> System.NullReferenceException: Object reference not set to an instance of an object.
   at Umbraco.Forms.Core.Services.FieldTypeStorage.GetFieldTypeByField(Field field)
   at Umbraco.Forms.Core.Data.Storage.RecordFieldValueStorage.GetRecordFieldValues(RecordField recordFieldInForm)
   at Umbraco.Forms.Core.Data.Storage.RecordFieldStorage.GetRecordFields(IEnumerable`1 recordFields, Form form)
   at Umbraco.Forms.Core.Data.Storage.RecordFieldStorage.GetAllRecordFields(IEnumerable`1 records, Form form)
   at Umbraco.Forms.Core.Data.Storage.RecordStorage.GetAllRecords(Form form, Boolean includeFields)
   at Umbraco.Forms.Core.Services.RecordService.GetAllRecords(Form form, Boolean includeFields)
   at Umbraco.Forms.Core.Data.RecordIndex.FormsIndexPopulator.PopulateIndexes(IReadOnlyList`1 indexes)
   at System.Threading.Tasks.Parallel.<>c__DisplayClass17_0`1.<ForWorker>b__1()
   at System.Threading.Tasks.Task.InnerInvokeWithArg(Task childTask)
   at System.Threading.Tasks.Task.<>c__DisplayClass176_0.<ExecuteSelfReplicating>b__0(Object )
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
   at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at System.Threading.Tasks.Parallel.ForWorker[TLocal](Int32 fromInclusive, Int32 toExclusive, ParallelOptions parallelOptions, Action`1 body, Action`2 bodyWithState, Func`4 bodyWithLocal, Func`1 localInit, Action`1 localFinally)
   at System.Threading.Tasks.Parallel.ForEachWorker[TSource,TLocal](IEnumerable`1 source, ParallelOptions parallelOptions, Action`1 body, Action`2 bodyWithState, Action`3 bodyWithStateAndIndex, Func`4 bodyWithStateAndLocal, Func`5 bodyWithEverything, Func`1 localInit, Action`1 localFinally)
   at System.Threading.Tasks.Parallel.ForEach[TSource](IEnumerable`1 source, Action`1 body)
   at Umbraco.Examine.IndexRebuilder.RebuildIndexes(Boolean onlyEmptyIndexes) in D:\a\1\s\src\Umbraco.Examine\IndexRebuilder.cs:line 55
   at Umbraco.Web.Search.BackgroundIndexRebuilder.RebuildOnStartupTask.Run() in D:\a\1\s\src\Umbraco.Web\Search\BackgroundIndexRebuilder.cs:line 93
---> (Inner Exception #0) System.NullReferenceException: Object reference not set to an instance of an object.
   at Umbraco.Forms.Core.Services.FieldTypeStorage.GetFieldTypeByField(Field field)
   at Umbraco.Forms.Core.Data.Storage.RecordFieldValueStorage.GetRecordFieldValues(RecordField recordFieldInForm)
   at Umbraco.Forms.Core.Data.Storage.RecordFieldStorage.GetRecordFields(IEnumerable`1 recordFields, Form form)
   at Umbraco.Forms.Core.Data.Storage.RecordFieldStorage.GetAllRecordFields(IEnumerable`1 records, Form form)
   at Umbraco.Forms.Core.Data.Storage.RecordStorage.GetAllRecords(Form form, Boolean includeFields)
   at Umbraco.Forms.Core.Services.RecordService.GetAllRecords(Form form, Boolean includeFields)
   at Umbraco.Forms.Core.Data.RecordIndex.FormsIndexPopulator.PopulateIndexes(IReadOnlyList`1 indexes)
   at System.Threading.Tasks.Parallel.<>c__DisplayClass17_0`1.<ForWorker>b__1()
   at System.Threading.Tasks.Task.InnerInvokeWithArg(Task childTask)
   at System.Threading.Tasks.Task.<>c__DisplayClass176_0.<ExecuteSelfReplicating>b__0(Object )<---
```

**Comment by nul800sebastiaan** [comment]
What version is this @Shazwazza? This should've been fixed in 8.4.1: https://github.com/umbraco/Umbraco.Forms.Issues/issues/339

**Comment by Shazwazza** [comment]
@nul800sebastiaan ah great, well this is a customer's site so i'm not entirely sure but seeing this in their logs. Unfortunately it prevents all indexes from continuing to populate so I'm fixing that part in the CMS.

**Comment by nul800sebastiaan** [comment]
I'll close this as being fixed in Umbraco 8.6.4 (through PR https://github.com/umbraco/Umbraco-CMS/pull/8398) as well as this shouldn't happen in the latest version of Forms any more.
